sdk/identity/identity/src/client/identityClient.ts

```diff
@@ -188,11 +200,60 @@ export class IdentityClient extends ServiceClient implements INetworkModule {
     }
   }
 
+  // Here is a custom layer that allows us to abort requests that go through MSAL,
+  // since MSAL doesn't allow us to pass options all the way through.
+
+  private abortControllers?: Record<string, AbortController[]>;
+
+  generateAbortSignal(correlationId?: string): AbortSignalLike {
+    if (!this.abortControllers) {
+      this.abortControllers = {};
+    }
+
+    const controller = new AbortController();
+    const key = correlationId || "noCorrelationId";
+
+    if (!this.abortControllers[key]) {
+      this.abortControllers[key] = [];
+    }
+
+    this.abortControllers[key].push(controller);
+    return controller.signal;
+  }
+
+  abortRequests(correlationId?: string): void {
+    const key = correlationId || "noCorrelationId";
+    let controller: AbortController | undefined;
+    while ((controller = this.abortControllers?.[key]?.shift())) {
+      controller.abort();
```

Comment:
even after aborting them, we don't throw away the controllers?

Reply:
they should get recycled automatically since I'm using shift. What doesn't get cleaned up is the index with an empty array.